config/src/main/java/org/axonframework/config/EventProcessingModule.java

```diff
@@ -869,6 +872,26 @@ public EventProcessingConfigurer registerPooledStreamingEventProcessorConfigurat
         return this;
     }
 
+    @Override
+    public EventProcessingConfigurer registerDeadLetterProvider(
+            BiFunction<Configuration, String, SequencedDeadLetterQueue<EventMessage<?>>> deadLetterProvider) {
+        this.deadLetterProviders.add(deadLetterProvider);
+        return this;
+    }
+
+    @Override
+    public Function<Configuration, SequencedDeadLetterQueue<EventMessage<?>>> provideDlq(String processingGroup){
+        return conf -> {
+            for(BiFunction<Configuration, String, SequencedDeadLetterQueue<EventMessage<?>>> provider : deadLetterProviders){
```

Comment:
I somewhat expected the `processingGroup` parameter of the `BiFunction` to be used as the filter to deduce if a DLQ should be constructed yes/no.